### PR TITLE
Add multipass spec

### PR DESCRIPTION
**Pull request by jsannemo**

### CODE REVIEW

File: spec/2023-07-draft/output_validators.md [hunk 1]
```diff
@@ -128,6 +128,16 @@ Note that a validator may choose to ignore the feedback directory
 entirely. In particular, the contest control system must not assume that
 the validator program creates any files there at all.
 
+<s class="kattis">
+### Multi-pass validation
+
+A multi-pass validator can be used for problems that should run the submission multiple times sequentially, using a new input generated by output validator during the previous invocation of the submission.
+
+To signal that the submission should be run once more, the output validator must exit with code 42 and output the new input in the file "nextpass" in the feedback directtory.
```
Comment (eldering) [inline spec/2023-07-draft/output_validators.md:136]: Nitpick:
```suggestion
To signal that the submission should be run once more, the output validator must exit with code 42 and output the new input in the file `nextpass.in` in the feedback directtory.
```

File: spec/2023-07-draft/output_validators.md [hunk 2]
```diff
@@ -128,6 +128,16 @@ Note that a validator may choose to ignore the feedback directory
 entirely. In particular, the contest control system must not assume that
 the validator program creates any files there at all.
 
+<s class="kattis">
```
Comment (niemela) [inline spec/2023-07-draft/output_validators.md:131]:
```suggestion
<div class="kattis">
```

File: spec/2023-07-draft/output_validators.md [hunk 3]
```diff
@@ -128,6 +128,16 @@ Note that a validator may choose to ignore the feedback directory
...
+### Multi-pass validation
+
+A multi-pass validator can be used for problems that should run the submission multiple times sequentially, using a new input generated by output validator during the previous invocation of the submission.
+
+To signal that the submission should be run once more, the output validator must exit with code 42 and output the new input in the file "nextpass" in the feedback directtory.
+
+It is a judge error to create the "nextpass" file and exit with any other code than 42.
+</s>
```
Comment (niemela) [inline spec/2023-07-draft/output_validators.md:139]:
```suggestion
</div>
```

**Review by niemela — CHANGES_REQUESTED** [review]

**Review by niemela — APPROVED** [review]

**Review by eldering — APPROVED** [review]

File: spec/2023-07-draft/output_validators.md [hunk 4]
```diff
@@ -128,6 +128,16 @@ Note that a validator may choose to ignore the feedback directory
 entirely. In particular, the contest control system must not assume that
 the validator program creates any files there at all.
 
+<div class="not-icpc">
+### Multi-pass validation
+
+A multi-pass validator can be used for problems that should run the submission multiple times sequentially, using a new input generated by output validator during the previous invocation of the submission.
+
+To signal that the submission should be run once more, the output validator must exit with code 42 and output the new input in the file `nextpass.in` in the feedback directtory.
```
Comment (Tagl) [inline spec/2023-07-draft/output_validators.md:136]:
```suggestion
To signal that the submission should be run once more, the output validator must exit with code 42 and output the new input in the file `nextpass.in` in the feedback directory.
```

File: spec/2023-07-draft/output_validators.md [hunk 5]
```diff
@@ -128,6 +128,16 @@ Note that a validator may choose to ignore the feedback directory
...
+
+It is a judge error to create the "nextpass" file and exit with any other code than 42.
```
Comment (Tagl) [inline spec/2023-07-draft/output_validators.md:138]:
```suggestion
It is a judge error to create the `nextpass.in` file and exit with any other code than 42.
```

File: spec/2023-07-draft/output_validators.md (hunk 4 above)
Comment (Tagl) [inline spec/2023-07-draft/output_validators.md]: I think you should clarify that the final pass is when the validator does not write a `nextpass.in`

**Review by Tagl — CHANGES_REQUESTED** [review]

**Review by Tagl — APPROVED** [review]

**Comment by thorehusfeldt** [comment]
The valid entries in `problem.yaml/validate` are now… what?

```cue
validation?: *"default" | "custom" | "interactive" | "score" | "interactive score" | "multipass" | "multipass score"
```
I wonder if there redundancy with 
```cue
 type?: *"pass-fail" | "scoring"
```
so that maybe `validation` could be just `"default" | "custom" | "interactive" | "multipass"`? (Honest question.)

**Comment by simonlindholm** [comment]
It's possible to have multi-pass interactive problems.

**Comment by jsannemo** [comment]
No, having custom scoring of a test case and having custom output validation are orthogonal concerns.

One could argue that "scoring" is unnecessary: we could just have the existence of "score.txt" imply this. But it's nice to catch the bug of a validator not outputting a score.

**Review by RagnarGrootKoerkamp — APPROVED** [review]

**Comment by thorehusfeldt** [comment]
Could you speculate about a richer syntax instead of just having a string of keywords? For instance, a YAML map like this:
```cue
validation:
  passes?: *1 | int | "unbounded"
  interactive?: *false | true
  score: *false | true
  validator: *"default" | "custom"
```
or whatever seems convenient to you.

**Pull request merged**